datafusion/core/src/execution/context/mod.rs

```diff
@@ -2860,6 +2863,57 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_register_default_functions() -> Result<()> {
+        let config = SessionConfig::new();
```

Comment:
While I understand the proposal to test with code like `SessionState::new` , I think it is problematic because if SessionState::new is changed, then this test will no longer match what it is doing.

I think the core of this test is verifying that there are no duplicate names in the alias lists.

Thus, perhaps we could write tests for `all_default_functions`, etc like

```rust
let mut names = HashSet::new();
for func in all_default_functions() {
  assert(names.insert(func.name()).is_none(), "func.name duplicated")
  for alias in func.aliases() {
    assert(names.insert(alias).is_none(), "alias duplicated")
   }
}
```

What do you think?

Reply:
Good idea. I'll add this test and remove the test for SessionState::new. Thanks